### PR TITLE
Make minor changes to OasGenerator

### PR DESCRIPTION
**Pull request by nimrod-up9**
* Added log message
* Remove Reset function from OasGenerator interface, use Stop+Start instead
* SetEntriesQuery returns a bool stating whether the query changed

### CODE REVIEW

File: agent/pkg/oas/oas_generator.go [hunk 1]
```diff
@@ -160,24 +159,28 @@ func (g *defaultOasGenerator) getGen(dest string, urlStr string) *SpecGen {
 	val, found := g.serviceSpecs.Load(dest)
 	var gen *SpecGen
 	if !found {
-		gen = NewGen(u.Scheme + "://" + dest)
+		serviceAddress := u.Scheme + "://" + dest
+		logger.Log.Infof("Create SpecGen for service %s", serviceAddress)
```
Comment (RoyUP9) [inline agent/pkg/oas/oas_generator.go:163]: I don't think we need this log, it could be spamming

**Comment by RoyUP9** [comment]
I don't see the change from reset to stop in the controller

File: agent/pkg/oas/oas_generator.go (hunk 1 above)
Comment (nimrod-up9) [reply agent/pkg/oas/oas_generator.go:163]: https://github.com/up9inc/mizu-enterprise/pull/137/commits/29a819e512052c3d982da7fb82f5fdd5b48b51a3

**Comment by nimrod-up9** [comment]
> I don't see the change from reset to stop in the controller

It's in Enterprise.
https://github.com/up9inc/mizu-enterprise/pull/137/commits/29a819e512052c3d982da7fb82f5fdd5b48b51a3

**Pull request merged**